ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/TlsConfiguration.scala

```diff
@@ -3,70 +3,166 @@
   def server: Option[SslContext] =
     if (enabled) {
-      scala.util.Using.resources(
+
+      val tlsInfo = scala.util.Using.resources(
         keyCertChainInputStreamOrFail,
         keyInputStreamOrFail,
       ) { (keyCertChain: InputStream, key: InputStream) =>
-        Some(
-          GrpcSslContexts
-            .forServer(
-              keyCertChain,
-              key,
-            )
-            .trustManager(trustCertCollectionFile.orNull)
-            .clientAuth(clientAuth)
-            .protocols(if (protocols.nonEmpty) protocols.asJava else null)
-            .build
+        val protocols = null.asInstanceOf[lang.Iterable[String]]
+        val defaultSslContext = buildServersSslContext(
+          keyCertChain = keyCertChain,
+          key = key,
+          protocols = protocols,
         )
+        TlsInfo.fromSslContext(defaultSslContext)
+      }
 
+      scala.util.Using.resources(
+        keyCertChainInputStreamOrFail,
+        keyInputStreamOrFail,
+      ) { (keyCertChain: InputStream, key: InputStream) =>
+        val protocols = protocolsNames(tlsInfo)
+        val sslContext = buildServersSslContext(
+          keyCertChain = keyCertChain,
+          key = key,
+          protocols = protocols,
+        )
+        logTlsProtocolsAndCipherSuites(sslContext, isServer = true)
+        Some(sslContext)
       }
-    } else None
+
+    } else {
+      logger.info(s"Server's TLS: Disabled.")
+      None
+    }
+
+  private def buildServersSslContext(
+      keyCertChain: InputStream,
+      key: InputStream,
+      protocols: lang.Iterable[String],
+  ) = {
+    GrpcSslContexts
+      .forServer(
+        keyCertChain,
+        key,
+      )
+      .trustManager(trustCertCollectionFile.orNull)
+      .clientAuth(clientAuth)
+      .protocols(protocols)
+      .sslProvider(SslContext.defaultServerProvider())
+      .build()
+  }
+
+  private[tls] def logTlsProtocolsAndCipherSuites(
+      sslContext: SslContext,
+      isServer: Boolean,
+  ): Unit = {
+    val who = if (isServer) "server" else "client"
+    val tlsInfo = TlsInfo.fromSslContext(sslContext)
+    logger.info(s"TLS $who: Enabled.")
+    logger.debug(s"TLS $who: Supported protocols: ${tlsInfo.supportedProtocols.mkString(", ")}.")
+    logger.info(s"TLS $who: Enabled protocols: ${tlsInfo.enabledProtocols.mkString(", ")}.")
+    logger.debug(
+      s"TLS $who: Supported cipher suites: ${tlsInfo.supportedCipherSuites.mkString(", ")}."
+    )
+    logger.info(s"TLS $who: Enabled cipher suites: ${tlsInfo.enabledCipherSuites.mkString(", ")}.")
+  }
 
   /** This is a side-effecting method. It modifies JVM TLS properties according to the TLS configuration. */
   def setJvmTlsProperties(): Unit =
     if (enabled && enableCertRevocationChecking) OcspProperties.enableOcsp()
 
+  private[tls] def protocolsNames(tlsInfo: TlsInfo): java.lang.Iterable[String] = {
```

Comment:
```suggestion
  private[tls] def filterSupportedProtocols(tlsInfo: TlsInfo): java.lang.Iterable[String] = {
```
Use a verb in the function name that describes what is being done to the list of the possible protocols

Reply:
Ok